Packages/ClientRuntime/Sources/Waiters/WaiterSequence.swift

```diff
@@ -0,0 +1,22 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// Enables a wait operation to be expressed as a sequence of asynchronous events.
+public struct WaiterSequence<Input, Output>: AsyncSequence {
```

Comment:
Do we need to define a public init so that SDK consumers can create instance of `WaiterSequence` if need be?